src/clj/game/cards/hardware.clj

```diff
@@ -1555,6 +1555,7 @@
                                    {:card card
                                     :replace-access
                                     {:mandatory true
+                                     :async true
                                      :effect (req (when (empty? (get-cards-to-access state))
```

Comment:
Seeing as we're making the ability async, this should be an `if` and the else branch calls `effect-completed`